src/camp/generators/lib/campch.py

```diff
@@ -150,6 +151,18 @@ def as_text(ari:ari.ARI) -> str:
         enc.encode(ari, buf)
         return buf.getvalue()
 
+    def as_timeperiod(value:datetime) -> str:
+      seconds = value.replace(tzinfo=timezone.utc).timestamp()
+      tv_sec = int(seconds)
+      tv_nsec = int(str(seconds).split(".")[1]) if "." in str(seconds) else 0 
+      return f"{{{tv_sec}, {tv_nsec}}}"
```

Comment:
This does not work properly; it is probably easiest to use the days, seconds, microseconds [timedelta](https://docs.python.org/3/library/datetime.html#datetime.timedelta) fields directly and convert with integer arithmetic.

Reply:
Updated the logic to properly convert fractional seconds to nanoseconds.